core/src/main/scala/sttp/client/ResponseAs.scala

```diff
@@ -142,8 +142,7 @@ object ResponseAs {
       }
 }
 
-sealed abstract class ResponseError[+T] extends Exception {
-  def body: String
-}
-case class HttpError(body: String) extends ResponseError[Nothing]
-case class DeserializationError[T](body: String, error: T) extends ResponseError[T]
+sealed abstract class ResponseError[+HE,+DE] extends Exception 
+case class HttpError[E](body: E) extends ResponseError[E,Nothing]
+case class UnknownHttpError(body: String) extends ResponseError[Nothing, Nothing]
```

Comment:
maybe we could try reducing the number of cases, removing `UnknownHttpError`. Then, the user would always have to provide total mapping for errors (e.g. using specific status codes + general case)

That's because in the default case of `ResponseError[E, String]`, we would have `HttpError(String)` which would occur on non-2xx responses, and `UnknownHttpError(String)` which would never occur

Reply:
Current implementation returns `UnknownHttpError` if there is no mapping for given status code ( also if mappings are empty). 

Although I am in favor of removing `UnknownHttpError` I am not sure if it is possible because that implies that `HttpError[String]` and `HttpError[MyCustomError]` have a common denominator. 

Specifying generic type as `Any` won't work because we will lose ability to perform exhaustive match.

Reply:
Yes, that would require that mappings be total (or rather, so that they include a fallback case for any non-covered status codes). If we have mappings at all, that is - by default we would just use `HttpError[String]`